src/Illuminate/Validation/Validator.php

```diff
@@ -601,10 +601,14 @@ protected function attributesThatHaveMessages()
         $results = [];
 
         foreach ($this->messages()->toArray() as $key => $message) {
-            $results[] = explode('.', $key)[0];
+            $key = explode('.', $key)[0];
```

Comment:
If we're going to optimize this, please pass a 3rd paramater to this function `1`. :)